### ISSUE TITLE
Parsing support for postgres `at time zone`

### ISSUE THREAD

**Issue by rabidaudio** [sqlfluff/sqlfluff#1331]
Hello folks, Postgres's `at time zone` conversion syntax seems to throw a parser error. Seems like `with time zone`/`without time zone` also. Postgres doesn't seem to have a function call version so I believe this is the only way to do it.

`example.sql`

```sql
SELECT ('2015-10-24 16:38:46'::timestamp AT TIME ZONE 'UTC');
```

## Observed Behaviour

```
==== sqlfluff ====
sqlfluff:                0.6.4 python:                 3.7.11
implementation:        cpython dialect:              postgres
verbosity:                   1 templater:               jinja

==== readout ====

=== [ path: example.sql ] ===

== [example.sql] FAIL
L:   1 | P:  41 |  PRS | Line 1, Position 41: Found unparsable section: " AT TIME
                       | ZONE 'UTC'"
==== summary ====
violations:        1 status:         FAIL
All Finished 📜 🎉!
```

## Steps to Reproduce

```bash
sqlfluff lint --dialect postgres -v example.sql 
```
